packages/debug/src/node/debug-adapter-session.ts

```diff
@@ -92,10 +85,18 @@ export class DebugAdapterSessionImpl implements DebugAdapterSession {
     }
 
     protected write(message: string): void {
-        this.communicationProvider.send(message);
+        if (!this.isClosed) {
+            this.debugAdapter.send(message);
+        }
     }
 
     async stop(): Promise<void> {
-        this.toDispose.dispose();
+        console.debug(`stopping debug adapter session: '${this.id}'`);
+
+        if (!this.isClosed) {
+            await this.debugAdapter.stop();
+        }
+        this.channel?.close();
+        this.channel = undefined;
     }
```

Comment:
The logic looks weird here: Shouldn't the channel track its own opened/closed state rather than storing it on the side like this (`this.isClosed`)?

Reply:
Weird it may be, but I'd rather not change state management, because that would require retesting.

Reply:
As a follow up then.